src/main/groovy/com/twcable/grabbit/jcr/JcrNodeDecorator.groovy

```diff
@@ -136,6 +136,41 @@ class JcrNodeDecorator {
     }
 
     /**
+     * Finds out nearest versionable ancestor for a node and performs a checkout
+    */
+    void checkoutNode() {
+        try {
+            JcrNodeDecorator decoratedVersionableAncestor = findVersionableAncestor();
+            if (decoratedVersionableAncestor && !decoratedVersionableAncestor.isCheckedOut()) {
+                decoratedVersionableAncestor.session.workspace.versionManager.checkout(decoratedVersionableAncestor.path);
+            }
+        }
+        catch (Exception exception) {
+            log.warn "Could not checkout node ${this.path}, ${exception.message}"
+        }
+    }
+
+    private JcrNodeDecorator findVersionableAncestor() throws RepositoryException {
+        if (isVersionable()) {
+            return this;
+        }
+        try {
+            JcrNodeDecorator parentDecoratedNode = new JcrNodeDecorator(this.parent);
+            return parentDecoratedNode.findVersionableAncestor();
+        } catch (ItemNotFoundException e) {
+            return null;
+        }
+    }
+
+    /**
+     * mix:simpleVersionable cannot be checked in. Throws javax.jcr.UnsupportedRepositoryOperationException
+     * Checking here for mix:versionable only
+     * */
+    private boolean isVersionable() throws RepositoryException {
```

Comment:
Minor, but I would remove the RepositoryException declaration here. Almost all JCR operations <i>potentially</i> through this exception as a catch all "something went wrong" in the vaguest of ways.   From our perspective, it's just sort of noise - We can't recover, and we can't really learn directly from it.